code/crates/core-consensus/src/handle/vote.rs

```diff
@@ -76,17 +76,14 @@ where
 
     debug_assert_eq!(consensus_height, vote_height);
 
-    // Only append to WAL and store the non-nil precommit if we have not yet seen this vote.
-    if !state.driver.votes().has_vote(&signed_vote) {
```

Comment:
Let's not remove this condition here, we want to avoid storing duplicate votes in the WAL for performance reasons: updating the WAL is expensive because it will issue a `fsync` syscall on the WAL file before consensus can resume and those are notoriously slow and expensive.